regtests/bin/matrix_cmake_ncep

```diff
@@ -109,6 +160,19 @@ EOF
     echo 'export KMP_STACKSIZE=2G'                          >> matrix.head
     echo 'export FI_OFI_RXM_BUFFER_SIZE=128000'             >> matrix.head
     echo 'export FI_OFI_RXM_RX_SIZE=64000'                  >> matrix.head
```

Comment:
@JessicaMeixner-NOAA these are retained in for orion, but not present for hercules.  Also, their position for orion has been moved above the module loads.  Are these both intentional?

Reply:
I actually don't see these being used in ufs-weather-model submit scripts.  I'm not sure why these are here although the `ulimit -s unlimited`  is usually needed.

Reply:
Those three were originally recommended on WCOSS2 when I was doing testing with SCOTCH a bit ago.  They were then also added to orion's job card when there were issues updating modules at a previous time and the issues seemed to go away.  I suspect we may need to include them on WCOSS2 when that update happens.  We have had all the `export`'s together at after the module loads, I don't know whether ordering matters with modules/exports, but I'd vote to choose one or the other to keep all job cards consistent between platforms 

I find it convenient to have the `cd` to regtests directory right after the `#SBATCH`s.  Could we keep that location?

Reply:
I'll look into this tomorrow.

Reply:
@MatthewMasarik-NOAA in my comparions  of what is happening in the develop branch on orion and what is happening in this branch in terms of the lodation of #sbatch and the "cd regtests" it is the same.  Is this different behavior for you?   

Since the hercules tests are working fine without extra envrionement variables beyond the unlimit -s, I prefer to not add anything.  We could add the environement variable to remove the one warning, but I'm okay leaving things as is.   I'll re-run a test on hercules to see if there are any system changes causing errors for me after my successful tests earlier this week/late last week.

Reply:
@JessicaMeixner-NOAA, to tie up a loose end regarding the "cd regtests".  Checking now the job card for `hera`, I see it doesn't have either the extra `export`'s or those `ulimit` lines, so the "cd regtests" does come directly after the Slurm directives, as I had in mind.  However, for `orion`, these lines **are** currently above the "cd regtests", so I was wrong on that.  Since nothing changed, no fix needed.

Reply:
Thanks @MatthewMasarik-NOAA   

I have tested that we can remove the extra orion variables for our regression tests.  Orion is a bit unstable right now so we should re-test that when we have a resolution to https://github.com/NOAA-EMC/WW3/issues/1152 and are ready to re-test this PR.